packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -501,4 +487,113 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
       throw new WalletSignMessageAndVerifyError(errMsg).message;
     }
   }
+
+  async prepRawTransaction(input: RawTransactionPrepPayload): Promise<AnyRawTransaction> {
+    try {
+      this.doesWalletExist();
+    } catch (error: any) {
+      const errMsg =
+          typeof error == "object" && "message" in error ? error.message : error;
+      throw new WalletSignTransactionError(errMsg).message;
+    }
+    if (!this._account?.address) {
+      throw new WalletNotSupportedMethod(`Wallet not providing address of account`).message;
+    }
+
+    const provider = this.getClient();
+
+    // If there is a fee payer, then it's a fee payer transaction
+    // Default expiration time is 60 seconds because there are more actions
+    const expiration_timestamp_secs = Math.floor(Date.now() / 1000) + (input.options?.expiration_milliseconds ?? 60000);
+    const additional_signers = input.additionalSigners?.map((value) => {return value.address}) ?? [];
+
+    // TODO: Add simulation for gas estimation
+    if (!input.feePayer && !input.additionalSigners) {
+      return await provider.generateTransaction(HexString.ensure(this._account.address), input.payload)
+    } else if (input.feePayer) {
+      return await provider.generateFeePayerTransaction(this._account.address, input.payload, input.feePayer.address, additional_signers, {
+        max_gas_amount: input.options?.max_gas_amount?.toString(),
+        gas_unit_price: input.options?.gas_unit_price?.toString(),
+        expiration_timestamp_secs: expiration_timestamp_secs.toString()
+      })
+    } else {
+      // TODO: Support multiagent without fee payer
+      throw new WalletNotSupportedMethod(`Multiagent without fee payer is not supported yet`).message;
+    }
+  }
+
+  async signRawTransaction(
+      rawTransaction: AnyRawTransaction
+  ): Promise<SignRawTransactionResponse | null> {
+    if (this._wallet && !("signRawTransaction" in this._wallet)) {
+      throw new WalletNotSupportedMethod(
+          `SignRawTransaction is not supported by ${this.wallet?.name}`
+      ).message;
+    }
+    try {
+      this.doesWalletExist();
+      const response = await (this._wallet as any).signRawTransaction(
+          rawTransaction
+      );
+      return response;
+    } catch (error: any) {
+      const errMsg =
+          typeof error == "object" && "message" in error ? error.message : error;
+      throw new WalletSignTransactionError(errMsg).message;
+    }
+  }
+
+  async signAndSubmitRawTransaction(input: RawTransactionRequest): Promise<any> {
+    if (this._wallet && !("signAndSubmitRawTransaction" in this._wallet)) {
+      throw new WalletNotSupportedMethod(`signAndSubmitRawTransaction not supported by ${this.wallet?.name}`).message;
+    }
+    try {
+      this.doesWalletExist();
+      return await (this._wallet as any).signAndSubmitRawTransaction(input);
+    } catch (error: any) {
+      throw this.convertError(error);
+    }
+  }
+
+  private getClient(): Provider {
```

Comment:
what is the need for this function?

Reply:
It's used above.

Basically, the sequence number and chain ID need to be pulled prior to generating the transaction.